silk/tests/openthread/ot_test_meshcop_joiner_discerner.py

```diff
@@ -0,0 +1,177 @@
+                                                     2).split("=")[1].strip()[1:-1]
+        self.assertTrue(leader_node_type == "leader", "Leader is not created correctly!!!")
+
+    @testcase.test_method_decorator
+    def test02_verify_discerner_value_after_device_reset(self):
+        # Verify Discerner value after device reset on Joiner.
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == 0)
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == 0)
+
+    @testcase.test_method_decorator
+    def test03_set_joiner_discerner_on_joiner1_and_joiner2(self):
+        # Set Joiner Discerner on `joiner1` and `joiner2`.
+        self.joiner1.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, str(D_LEN1))
+        self.joiner1.set(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE, DISCERNER1)
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == D_LEN1)
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == int(DISCERNER1, 0))
+
+        self.joiner2.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, str(D_LEN2))
+        self.joiner2.set(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE, DISCERNER2)
+        verify(int(self.joiner2.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == D_LEN2)
+        verify(int(self.joiner2.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == int(DISCERNER2, 0))
+
+    @testcase.test_method_decorator
+    def test04_verify_clearing_of_previously_set_joiner_discerner(self):
+        # Set Joiner Discerner on `joiner3`.
+        self.joiner3.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, str(D_LEN2))
+        self.joiner3.set(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE, DISCERNER2)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == D_LEN2)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == int(DISCERNER2, 0))
+
+        # Clear Joiner Discerner on `joiner3`.
+        self.joiner3.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, '0')
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == 0)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == 0)
+
+    @testcase.test_method_decorator
+    def test05_add_joiners_on_commissioner(self):
+        self.commissioner.commissioner_start()
+        self.wait_for_completion(self.device_list)
+
+        # Add `joiner1` and `joiner2` using joiner discerner.
+        self.commissioner.commissioner_add_joiner_with_discerner(DISCERNER1, D_LEN1, PSK1, JOINER_TIMOUT)
+        self.commissioner.commissioner_add_joiner_with_discerner(DISCERNER2, D_LEN2, PSK2, JOINER_TIMOUT)
+
+        # Add `joiner3` using EUI64 Hardware Address of joiner3.
+        joiner3_hw_addr = self.joiner3.get(wpan.WPAN_HW_ADDRESS)[1:-1]  # Remove the `[]`
+        self.commissioner.commissioner_add_joiner(joiner3_hw_addr, PSK3, JOINER_TIMOUT)
```

Comment:
For adding joiner3 seeing joiner-add failure.
Please find below details of the error: 
---------
[2020-10-13 22:46:26,452] [silk.Dev-5B97] [DEBUG] sudo ip netns exec ttyACM6 /usr/local/bin/wpanctl -I ttyACM6 commissioner joiner-add F4CE361A35F7EB9E 500 THELAST0FUS
[2020-10-13 22:46:26,503] [silk.Dev-5B97.wpantund] [DEBUG] wpantund[17221]: SpinelNCPTaskSendCommand.cpp:351: Requirement Failed ((mRetVal) == 0)
[2020-10-13 22:46:26,504] [silk.Dev-5B97.wpantund] [DEBUG] wpantund[17221]: SendCommand task encountered an error: 15335435 (0x00EA000B)
[2020-10-13 22:46:26,506] [silk.Dev-5B97] [DEBUG] [stdout] commissioner joiner-add failed with error 15335435. NCP-Specific Errorcode

thanks

Reply:
This issue was due to missing flag  OPENTHREAD_CONFIG_COMMISSIONER_MAX_JOINER_ENTRIES in ot build.
Fixed it.